src/libcore/pin.rs

```diff
@@ -191,15 +297,46 @@ impl<P: Deref> Pin<P> {
     /// not guarantee that the data `P` points to is pinned, constructing a
```

Comment:
I think maybe we should re-iterate what "pinned" means in this context. Specifically, that the target's memory location will not change until it is dropped.